src/client/components/pages/collections.js

```diff
@@ -41,9 +41,16 @@ class CollectionsPage extends React.Component {
 
 	handleTypeChange(type) {
 		const querySearchParams = type ? `type=${type}` : '';
-		this.setState({querySearchParams});
+		this.setState({querySearchParams, type});
 	}
 
+	searchParamsChangeCallback = (searchParms) => {
+		const type = searchParms.get('type') ?? '';
+		if (type !== this.state.type) {
+			this.setState({querySearchParams: `?${searchParms.toString()}`, type});
+		}
```

Comment:
I forgot about this little detail, sorry:

setState has a lot of smarts built-in, and won't re-render the page if the values haven't changed. So you should be able to simply do this and not worry about whether type has changed, if the rest of the search paramas have changed or not, etc. :


```suggestion
		this.setState({querySearchParams: `?${searchParms.toString()}`, type});
```

Reply:
Just to be sure , won't it cause the PagerElement to render twice ? , like it will also run setState when querySearchParms changes ( cause of modified from ,size parameter) which change on next and back button.

Reply:
Fair enough, I'm not 100% sure of what would happen. 😅 
I don't think it would be a big impact, but might as well keep the condition here. :+1: